docs/rfcs/2024-03-19-external-services.md

```diff
@@ -0,0 +1,192 @@
+
+The examples consider the scenarios when SimpleX Chat is a consumer, and another app is a provider of the payment services.
+
+### Custom URL schemes
+
+This approach is the simplest to open some page or perform some action in the provider app. The downside of this approach is that if it's used on its own, then it may result in opening any other app, as any app can declare this custom scheme.
+
+Some other approach should be added to determine if the provider app is installed, for example App groups - see below. While security considerations can be addressed with the universal links, they have privacy downside, as they will result in undesirable Internet access in case the target app is not installed.
+
+For example, if the user wants to see the list of all transactions with the contact or any other detailed information about the payments, it may be better to use URLs to open it in the payment provider's UI, rather than to show it in SimpleX Chat UI.
+
+### Share extension
+
+This extension can be used when it is required to load some information from the provider app and provide some custom interface to perform some action.
+
+While it's commonly used to share content and to send messages, it can also be used to send the payments or request some other actions, when custom provider UI is appropriate and interaction is mostly one way.
+
+### App groups
+
+App group is a shared container that can be accessed by multiple apps or multiple parts on one app. Traditionally it is used for sharing data with the app extensions (e.g., Share or Notification service extensions) and for sharing data between the multiple apps of the same developers. It can also be used by sharing the information via files and also communicating between two different running processes via file coordinator notifications.
+
+While this approach can be used for sharing files and communication between active processes, it has these limitations:
+- the name of the app group requires some management, as it has to be common for two apps.
+- the suspended app will not receive any notifications.
+
+### File provider extension
+
+This extension allows sharing files and access to folders with other applications and processes. This extension can wake to process file access and change requests, and also can be used to execute arbitrary XPC service calls associated with files, so unlike app groups it will allow the app to process service requests from suspended state.
+
+File provider extension has these limitations:
+- it has to run within 25 mb memory (other extensions have the same limit).
+- it is a separate process, and unless the provider app already uses shared files and shared database it may require some re-engineering to move app data to shared container of the app's app group.
+- it has to use files, as while it's possible to use a "fake file" simply to create a service, it would not be compliant with app store policies.
```

Comment:
quickly searched policy but didn't find this specific place.. not sure masquerading api calls as files wouldn't constitute as fake files, if they specifically mean files in common meaning of user files.